### PR TITLE
Update starter workflows to specify required permissions

### PR DESCRIPTION
**Pull request by ericsciple**
Updating starter workflows in reaction to recent changes: [Control permissions for GITHUB_TOKEN](https://github.blog/changelog/2021-04-20-github-actions-control-permissions-for-github_token/)

## Pre-requisites

- [ ] Prior to submitting a new workflow, please apply to join the GitHub Technology Partner Program: [partner.github.com/apply](https://partner.github.com/apply?partnershipType=Technology+Partner).

---

### **Please note that at this time we are only accepting new starter workflows for Code Scanning. Updates to existing starter workflows are fine.**

---

## Tasks

**For _all_ workflows, the workflow:**

- [ ] Should be contained in a `.yml` file with the language or platform as its filename, in lower, [_kebab-cased_](https://en.wikipedia.org/wiki/Kebab_case) format (for example, [`docker-image.yml`](https://github.com/actions/starter-workflows/blob/main/ci/docker-image.yml)).  Special characters should be removed or replaced with words as appropriate (for example, "dotnet" instead of ".NET").
- [ ] Should use sentence case for the names of workflows and steps (for example, "Run tests").
- [ ] Should be named _only_ by the name of the language or platform (for example, "Go", not "Go CI" or "Go Build").
- [ ] Should include comments in the workflow for any parts that are not obvious or could use clarification.

**For _CI_ workflows, the workflow:**

- [ ] Should be preserved under [the `ci` directory](https://github.com/actions/starter-workflows/tree/main/ci).
- [ ] Should include a matching `ci/properties/*.properties.json` file (for example, [`ci/properties/docker-publish.properties.json`](https://github.com/actions/starter-workflows/blob/main/ci/properties/docker-publish.properties.json)).
- [ ] Should run on `push` to `branches: [ $default-branch ]` and `pull_request` to `branches: [ $default-branch ]`.
- [ ] Packaging workflows should run on `release` with `types: [ created ]`.
- [ ] Publishing workflows should have a filename that is the name of the language or platform, in lower case, followed by "-publish" (for example, [`docker-publish.yml`](https://github.com/actions/starter-workflows/blob/main/ci/docker-publish.yml)).

**For _Code Scanning_ workflows, the workflow:**

- [ ] Should be preserved under [the `code-scanning` directory](https://github.com/actions/starter-workflows/tree/main/ci).
- [ ] Should include a matching `code-scanning/properties/*.properties.json` file (for example, [`code-scanning/properties/codeql.properties.json`](https://github.com/actions/starter-workflows/blob/main/code-scanning/properties/codeql.properties.json)), with properties set as follows:
  - [ ] `name`: Name of the Code Scanning integration.
  - [ ] `organization`: Name of the organization producing the Code Scanning integration.
  - [ ] `description`: Short description of the Code Scanning integration.
  - [ ] `categories`: Array of languages supported by the Code Scanning integration.
  - [ ] `iconName`: Name of the SVG logo representing the Code Scanning integration. This SVG logo must be present in [the `icons` directory](https://github.com/actions/starter-workflows/tree/main/icons).
- [ ] Should run on `push` to `branches: [ $default-branch, $protected-branches ]` and `pull_request` to `branches: [ $default-branch ]`. We also recommend a `schedule` trigger of `cron: $cron-weekly` (for example, [`codeql.yml`](https://github.com/actions/starter-workflows/blob/c59b62dee0eae1f9f368b7011cf05c2fc42cf084/code-scanning/codeql.yml#L14-L21)).

**Some general notes:**

- [ ] This workflow must _only_ use actions that are produced by GitHub, [in the `actions` organization](https://github.com/actions), **or**
- [ ] This workflow must _only_ use actions that are produced by the language or ecosystem that the workflow supports.  These actions must be [published to the GitHub Marketplace](https://github.com/marketplace?type=actions).  We require that these actions be referenced using the full 40 character hash of the action's commit instead of a tag.  Additionally, workflows must include the following comment at the top of the workflow file:
    ```
    # This workflow uses actions that are not certified by GitHub.
    # They are provided by a third-party and are governed by
    # separate terms of service, privacy policy, and support
    # documentation.
    ```
- [ ] Automation and CI workflows should not send data to any 3rd party service except for the purposes of installing dependencies.
- [ ] Automation and CI workflows cannot be dependent on a paid service or product.

### CODE REVIEW

File: automation/manual.yml
Comment (ericsciple) [inline automation/manual.yml:26]: remove

File: ci/maven.yml
Comment (ericsciple) [inline ci/maven.yml:26]: i branched off the old default branch :(

Comment (Azrail16) [reply ci/maven.yml]: Serkan

**Comment by skedwards88** [comment]
@ericsciple Does the `automation/greetings.yml` workflow also need increased permissions? If not, can you help me understand why?

**Comment by hubwriter** [comment]
Do the code scanning starter workflows (https://github.com/actions/starter-workflows/tree/main/code-scanning) need updated to add:
```
  security-events: write
```

and possibly also (for codeql-action/init in codeql.yml):
```
  actions: read
```

File: code-scanning/codeql.yml
Comment (nickfyson) [inline code-scanning/codeql.yml:29]:
```suggestion
      security-events: write
      actions: read
```
@hubwriter is correct that we should include this additional permission for the codeql workflow.

**Review by nickfyson — APPROVED** [review]
@hubwriter is correct that we need an extra permission for the codeql workflow

Other than that, looks fine for all the `code-scanning` workflows! 👍

File: code-scanning/codeql.yml
Comment (ericsciple) [reply code-scanning/codeql.yml:30]: Thanks! Fixed

**Comment by ericsciple** [comment]
I will wait to merge until Monday, since I don't want to break anything on a Friday afternoon/evening :)

**Comment by nickfyson** [comment]
NB I only have any ownership/knowledge of the workflows under `code scanning`, so you'll probably want to get approval from the actions experience team as well. 🙂

**Comment by ylemkimon** [comment]
This may be out of scope for this PR, but I think it'd be nice to set only `contents: read` for workflows that don't need write access, such as https://github.com/actions/starter-workflows/blob/main/automation/manual.yml, and link to the documentation.

**Review by cschleiden — APPROVED** [review]

**Review by ServrBig4uNet — COMMENTED** [review]
Thank you

**Review by ServrBig4uNet — COMMENTED** [review]
Thanks

**Comment by ServrBig4uNet** [comment]
Thanks

**Pull request merged**
